### PR TITLE
* External references

### PR DESCRIPTION
**Pull request by ColonelSanderson**
This was also part of the first round of feedback.

### CODE REVIEW

File: backend/plugin_init.rb [hunk 1]
```diff
@@ -42,11 +42,11 @@
...
-DB.attempt {
-  RAP.get_default_id
-}.and_if_constraint_fails do |e|
-  Log.warn("Constraint failure while creating default RAP: #{e}")
-end
+# DB.attempt {
+#   RAP.get_default_id
+# }.and_if_constraint_fails do |e|
+#   Log.warn("Constraint failure while creating default RAP: #{e}")
+# end
```
Comment (payten) [inline backend/plugin_init.rb]: Was this causing you problems @ColonelSanderson?

**Comment by payten** [comment]
Just the one question above @ColonelSanderson.  Otherwise, looks great.

**Comment by ColonelSanderson** [comment]
Yes I need to do more testing but I think that it fails for fresh installs because it can't find a default repo id. (Though I didn't mean to commit the change)

**Comment by payten** [comment]
Ah boo, good to know. I'll look into it.

**Comment by payten** [comment]
Just pushed a patch @ColonelSanderson.  I've tinkered the code to create the RAP for existing repositories and when they are created.  If you rebase, hopefully bootstrapping a new DB should be less sad.

**Comment by ColonelSanderson** [comment]
Great, should be good to go.

**Comment by payten** [comment]
Thanks @ColonelSanderson.  Manually merged this one.

**Pull request closed without merging**